### DISCUSSION TITLE
Mapping the color distribution of a 3D point cloud to a volume

### DISCUSSION THREAD

**Discussion by Yao-14** [pyvista/pyvista#2264] [Q&A]
> 
Hi everyone, is there a way to map the color of the point cloud to the reconstructed volume in pyvista?
My idea is:
1. Based on the 3D point cloud, reconstruct the surface through pyvista.PolyDataFilters.reconstruct_surface()
2. Voxelize the reconstructed surface by pv.voxelize() to generate a volume
3. Add color information to the reconstructed volume and map the color of the point cloud to the reconstructed volume

The following two figures are the reconstructed voxelized point cloud and volumetric mesh. My question is how to map the color distribution on the point cloud to the volumetric mesh?
![pcd](https://user-images.githubusercontent.com/89448797/155438568-216f6c75-a2bf-4c0e-87c0-4f950a035237.png)
![volume](https://user-images.githubusercontent.com/89448797/155438572-26da81ad-d06a-495b-a792-e33fa3df4d55.png)
Hi everyone, is there a way to map the color of the point cloud to the reconstructed volume in pyvista?

**Comment by Yao-14** [comment]
The most basic code is as follows：
pcd = pv.PolyData(bucket_xyz)
surf = pcd.delaunay_3d().extract_surface()
volume = pv.voxelize(surf, density=surf.length / 200, check_surface=False)

**Comment by RichardScottOZ** [comment]
Do you want just the points colored on the volume, or the whole volume colored?

**Reply by Yao-14** [reply]
the whole volume colored， actually

**Comment by RichardScottOZ** [comment]
Use the colormap from the first part - set to the same limits if you want?

**Comment by banesullivan** [comment]
I think you need to interpolate the values of the point cloud onto the volume.

Reference https://docs.pyvista.org/examples/01-filter/interpolate.html

Something like:

```py
volume.interpolate(points, radius=...) 
```
